dataactcore/interfaces/function_bag.py

```diff
@@ -41,3 +46,39 @@ def getPasswordHash(password, bcrypt):
     hash = bcrypt.generate_password_hash(password + salt, HASH_ROUNDS)
     password_hash = hash.decode("utf-8")
     return salt, password_hash
+
+
+def populateSubmissionErrorInfo(submissionId):
+    """Set number of errors and warnings for submission."""
+    sess = GlobalDB.db().session
+    submission = sess.query(Submission).filter(Submission.submission_id == submissionId).one()
+    submission.number_of_errors = sumNumberOfErrorsForJobList(submissionId)
+    submission.number_of_warnings = sumNumberOfErrorsForJobList(submissionId, errorType='warning')
+    sess.commit()
+
+
+def sumNumberOfErrorsForJobList(submissionId, errorType='fatal'):
+    """Add number of errors for all jobs in list."""
+    sess = GlobalDB.db().session
+    errorSum = 0
+    jobs = sess.query(Job).filter(Job.submission_id == submissionId).all()
+    for job in jobs:
+        jobErrors = checkNumberOfErrorsByJobId(job.job_id, errorType)
+        if errorType == 'fatal':
+            job.number_of_errors = jobErrors
+        elif errorType == 'warning':
+            job.number_of_warnings = jobErrors
+        sess.add(job)
+        errorSum += jobErrors
+    sess.commit()
+    return errorSum
+
+
+def checkNumberOfErrorsByJobId(jobId, errorType='fatal'):
+    """Get the number of errors for a specified job and severity."""
+    sess = GlobalDB.db().session
+    errors = sess.query(
+        func.sum(ErrorMetadata.occurrences)).join(
+        ErrorMetadata.severity).filter(
+        ErrorMetadata.job_id == jobId, RuleSeverity.name == errorType).scalar()
+    return errors or 0
```

Comment:
Dig a bit more into this. The ErrorMetadata table actually tracks error/warning count by combination of job/filename/fieldname, and error type. So jobs w/o errors and warnings won't be represented, which is why the code returns a 0 when the query doesn't find anything.

To your great point about the new `sum` function potentially breaking on NULL values--the field being summed is not nullable. I also checked the code that inserts ErrorMetadata records...it always starts counting at 1. For these reasons, the refactor should work the same way as the old code.